### PR TITLE
[mobile_slides] Showing different slides in mobile version.

### PR DESCRIPTION
**Pull request by SilicorniO**
In mobile version horizontal banners are shown very small. For this reason I have added the possibility of define specific slides for mobile version. 

To do it I have added a property to the slide as "mobile". If there are not mobile slides, the rest of slides are shown, but if one mobile slide is found, then that slide is shown when mobile version of website is running. 

I didn't prepare it to be updated because I don't know how it works, it is the first time I modify a module. If you consider this feature is right and you will accept the pull request I can investigate and develop it.

I have it already working in [https://www.esenciacustome.com](https://www.esenciacustome.com) with default theme that I had to change the template too, but I can do another pull request to make it compatible if this is accepted.

Commit message: 

BO: Each slide has a new check for mobile version. If it is checked the mobile slider will show only these slides and they won't be shown in desktop or tablet version.
FO: Now there are two slides in the template. Only one is shown at same time. If there are not mobile slides, then all slides are shown in both types.
NOTE: To update it is necessary to add the field 'mobile' to the table ''PREFIX_homeslider_slides' of the database. Or uninstall and install again.

### CODE REVIEW

File: css/homeslider.css [hunk 1]
```diff
@@ -72,3 +72,22 @@
   background-position: right top;
   right: 0;
 }
+
+/* Show or hide mobile slider */
+@media (max-width: 767px) {
+  .homeslider-container-no-mobile {
+    display: none;
+  }
+	.homeslider-container-mobile {
```
Comment (Progi1984) [inline css/homeslider.css]:
```suggestion
  .homeslider-container-mobile {
```

File: css/homeslider.css [hunk 2]
```diff
@@ -72,3 +72,22 @@
...
+    display: block;
+  }
+}
+
+@media (min-width: 767px) {
+	.homeslider-container-no-mobile {
```
Comment (Progi1984) [inline css/homeslider.css]:
```suggestion
  .homeslider-container-no-mobile {
```

File: css/homeslider.css [hunk 3]
```diff
@@ -72,3 +72,22 @@
...
+    display: block;
+  }
+	.homeslider-container-mobile {
```
Comment (Progi1984) [inline css/homeslider.css]:
```suggestion
  .homeslider-container-mobile {
```

File: js/homeslider.js [hunk 4]
```diff
@@ -39,4 +39,21 @@ jQuery(document).ready(function ($) {
...
+    pause: $('.homeslider-container').data('pause'), // Boolean: Pause on hover, true or false
+    maxwidth: "",           // Integer: Max-width of the slideshow, in pixels
+    namespace: "homeslider",   // String: Change the default namespace used
+    before: function(){},   // Function: Before callback
+    after: function(){}     // Function: After callback
+  };
+
+  var rSlidesMobile = $(".rslides-mobile");
+  if (rSlidesMobile) {
+    $(".rslides-mobile").responsiveSlides(homesliderConfigMobile);
```
Comment (Progi1984) [inline js/homeslider.js]:
```suggestion
    rSlidesMobile.responsiveSlides(homesliderConfigMobile);
```

File: ps_imageslider.php [hunk 5]
```diff
@@ -432,6 +443,18 @@ protected function _postProcess()
...
             $this->clearCache();
             $this->_html .= ($res ? $this->displayConfirmation($this->getTranslator()->trans('Configuration updated', array(), 'Admin.Notifications.Success')) : $this->displayError($this->getTranslator()->trans('The configuration could not be updated.', array(), 'Modules.Imageslider.Admin')));
+        
+        } elseif (Tools::isSubmit('changeMobile') && Tools::isSubmit('id_slide')) {
+            $slide = new Ps_HomeSlide((int)Tools::getValue('id_slide'));
+            if ($slide->mobile == 0) {
+                $slide->mobile = 1;
+            } else {
+                $slide->mobile = 0;
+            }
```
Comment (Progi1984) [inline ps_imageslider.php]:
```suggestion
            $slide->mobile = $slide->mobile == 0 ? 1 : 0;
```

File: ps_imageslider.php [hunk 6]
```diff
@@ -632,21 +665,22 @@ public function getNextPosition()
...
+            SELECT hs.`id_homeslider_slides` as id_slide, hss.`position`, hss.`active`, hss.`mobile`, hssl.`title`,
             hssl.`url`, hssl.`legend`, hssl.`description`, hssl.`image`
             FROM '._DB_PREFIX_.'homeslider hs
             LEFT JOIN '._DB_PREFIX_.'homeslider_slides hss ON (hs.id_homeslider_slides = hss.id_homeslider_slides)
             LEFT JOIN '._DB_PREFIX_.'homeslider_slides_lang hssl ON (hss.id_homeslider_slides = hssl.id_homeslider_slides)
             WHERE id_shop = '.(int)$id_shop.'
             AND hssl.id_lang = '.(int)$id_lang.
-            ($active ? ' AND hss.`active` = 1' : ' ').'
+            ($active ? ' AND hss.`active` = 1' : ' ').
+            ($mobile !== null ? (' AND hss.`mobile` = '.($mobile? '1' : '0')) : ' ').'
```
Comment (Progi1984) [inline ps_imageslider.php]:
```suggestion
            ($mobile !== null ? (' AND hss.`mobile` = ' . ($mobile ? '1' : '0')) : ' ') . '
```

**Comment by SilicorniO** [comment]
Hi, I don't know if you need something else from myself. Else let me know.

**Comment by Progi1984** [comment]
@SilicorniO Hello, thank you for your contribution. I have commented your pull request with some feedback. Could you fix them ? and fix the conflicting file ? Thanks

**Comment by SilicorniO** [comment]
Hi Progi1984, I approved everything but I think there is a conflict to resolve but I don't have write permissions. The code to fix is:

`
<<<<<<< features/mobile_slides
        $slides = $this->getSlides(true, false);
        $this->getWidgetVariablesSlides($slides);

        $slidesMobile = $this->getSlides(true, true);
        $this->getWidgetVariablesSlides($slidesMobile);
=======
        $slides = $this->getSlides(true);
        if (is_array($slides)) {
            foreach ($slides as &$slide) {
                $slide['sizes'] = @getimagesize((__DIR__ . DIRECTORY_SEPARATOR . 'images' . DIRECTORY_SEPARATOR . $slide['image']));
                if (isset($slide['sizes'][3]) && $slide['sizes'][3]) {
                    $slide['size'] = $slide['sizes'][3];
                }
            }
        }
>>>>>>> dev`

The code to keep is:

`$slides = $this->getSlides(true, false);
        $this->getWidgetVariablesSlides($slides);

        $slidesMobile = $this->getSlides(true, true);
        $this->getWidgetVariablesSlides($slidesMobile);`

I created a new function called "getWidgetVariablesSlides" that contains part of the conflict code. 

If I have to change something manually, please tell me how I can do it. It is the first merge I launch to a public library in Github.

Thanks.

**Comment by RubixR4** [comment]
Hi there,

Do you have news about this feature ?

A lot of my customers mention the bad ratio of images when I activate the module for mobile devices. It would be perfect to upload different images for mobile. It is possible to correct the conflict and then publish a new version ?

Thanks !

**Comment by SilicorniO** [comment]
Hi RubixR4, I tried to launch a PR but there was a conflict and Git didn't show me it. Let me check if I can launch again the PR fixing the problem reported by Progi1984

**Comment by SilicorniO** [comment]
I'm sorry RubixR4, I've been checking and there is no conflict in my code or I can't see it. There is a button in this page to "resolve conflicts". It openes a page where there is a conflict that not exist in my code and I can't fix it because I don't have write access, so I can't fix it. It would be very easy, but I would need help to know How I can do it.

**Comment by RubixR4** [comment]
Hi @Progi1984, would it be possible to fix the conflict yourself please ? If it's not possible, could you help @SilicorniO to do so ?
It would be so great to have this feature in the next version... 

Thank you so much !

File: ps_imageslider.php (hunk 5 above)
Comment (PierreRambaud) [reply ps_imageslider.php:453]: I don't get it, so whatever you set, it's always 1/true/value :thinking:

**Pull request closed without merging**